pkg/kv/mem/store.go

```diff
@@ -37,23 +37,12 @@ func init() {
 
 func (d *Driver) Open(_ context.Context, _ kvparams.KV) (kv.Store, error) {
 	return &Store{
-		m:    make(map[string][]byte),
-		keys: []string{},
+		m: make(map[string]map[string]kv.Entry, 0),
```

Comment:
Looks like a comment explaining the first is the partition and the second is the key could be helpful.
Or better, IMHO, define a partition type that is a key->val mapping and then `m` which is a key->partition mapping.